www/src/components/semesters/SemesterCalendar.vue

```diff
@@ -55,24 +55,40 @@ function translate(text: keyof (typeof translations)[LangId]): string {
     return translations[langId][text];
 }
 
-const { semester, scheduleChanges, subjects, classrooms, allUsers } = defineProps<{
+const { semester, scheduleChanges, subjects, classrooms, allUsers, subjectGroups } = defineProps<{
     semester: SemesterData;
     scheduleChanges: ScheduleChangeData[];
     subjects: SubjectData[];
     classrooms: ClassroomData[];
     allUsers: UserPublicData[];
+    subjectGroups: { subjectsData: SubjectData[]; title: string }[];
 }>();
 
 const teachers = computed<UserPublicData[]>(() => {
     return allUsers.filter(user => subjects.some(subject => subject.teachers.some(teacher => teacher.id === user.id)));
 });
 
-const selectedSubjects = ref<SubjectData[]>([]);
 const selectedClassrooms = ref<ClassroomData[]>([]);
 const selectedTeachers = ref<UserPublicData[]>([]);
 
 const subjectColors = computed(() => Object.fromEntries(subjects.map(subject => [subject.id, subject.color])));
 
+const subjectGroupSelections = ref<Record<string, SubjectData[]>>({});
```

Comment:
What's the purpose of this grouping? It seems it's only grouped to be later flattened...

Reply:
Okay it's a workaround for ToggleButtonPicker not handling v-model correctly...